File: ALODWebUtility/Documents/DocumentProcessingFactory.vb
Imports ALOD.Core.Interfaces

Namespace Documents

    Public Class DocumentProcessingFactory

        Public Function GetCertificationStampStrategy() As IDocumentProcessingStrategy
            Return New CertificationStampStrategy()
        End Function

        Public Function GetDefaultStrategy() As IDocumentProcessingStrategy
            Return New DefaultProcessingStrategy()
        End Function

        Public Function GetStrategyByType(ByVal type As ProcessingStrategyType) As IDocumentProcessingStrategy
            Select Case type
                Case ProcessingStrategyType._Default
                    Return GetDefaultStrategy()

                Case ProcessingStrategyType.CertificationStamp
                    Return GetCertificationStampStrategy()

                Case Else
                    Return GetDefaultStrategy()
            End Select
        End Function

    End Class

End Namespace

File: ALODWebUtility.CS/Documents/DocumentProcessingFactory.cs
using ALOD.Core.Interfaces;

namespace ALODWebUtility.Documents
{
    public class DocumentProcessingFactory
    {
        public IDocumentProcessingStrategy GetCertificationStampStrategy()
        {
            return new CertificationStampStrategy();
        }

        public IDocumentProcessingStrategy GetDefaultStrategy()
        {
            return new DefaultProcessingStrategy();
        }

        public IDocumentProcessingStrategy GetStrategyByType(ProcessingStrategyType type)
        {
            switch (type)
            {
                case ProcessingStrategyType._Default:
                    return GetDefaultStrategy();

                case ProcessingStrategyType.CertificationStamp:
                    return GetCertificationStampStrategy();

                default:
                    return GetDefaultStrategy();
            }
        }
    }
}
